src/helpers.js

```diff
@@ -148,12 +151,25 @@ function mergeHashes(source, target, options = {}) {
         old[key] = nested.old
       }
     } else if (target[key] !== undefined) {
-      if (typeof source[key] === 'string' || Array.isArray(source[key])) {
-        target[key] = source[key]
-        mergeCount += 1
-      } else {
+      if (typeof source[key] !== 'string' && !Array.isArray(source[key])) {
         old[key] = source[key]
         oldCount += 1
+      } else {
+        if (
+          (options.resetOnChange &&
+            !isPlural(key) &&
+            typeof source[key] === 'string' &&
+            source[key] !== target[key]) ||
```

Comment:
I haven't read the whole source code, but does `source[key]`  hold the value of each key in the JSON file being parsed? If so, what does the `target[key]` hold?

Not sure if it might be worth adding some comments around the behavior or maybe even some tests?

Tests might help visualize how this is all working though.

Reply:
`source[key]` depends on the context `merge` is being called in, but in the main context it's the translated values in that locale/namespace, and `target[key]` is anything that was parsed out of the code (so default values), or `undefined` if it's absent.

Reply:
`target` is also the "end result", so in some cases the default values are left there and sometimes they are overwritten by the translated values.